File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Calendars/createstandardcalendar.rb
module Asposetasksjava
  module CreateStandardCalendar
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        # Define Calendar
    	cal1 = project.getCalendars().add("My Cal")
    	Rjb::import('com.aspose.tasks.Calendar').makeStandardCalendar(cal1)

    	# Save the Project
    	project.save(data_dir + "CreateStandardCalendar.xml", Rjb::import('com.aspose.tasks.SaveFileFormat').XML)

    	puts "Created standard calendar, please check the output file."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithCalendars/CreateStandardCalendar.php
<?php
namespace Aspose\Tasks\WorkingWithCalendars;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Calendar as Calendar;
use com\aspose\tasks\SaveFileFormat as SaveFileFormat;

class CreateStandardCalendar{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        # Define Calendar
    	$cal1 = $project->getCalendars()->add("My Cal");
        $calendar=new Calendar();
    	$calendar->makeStandardCalendar($cal1);

    	# Save the Project
        $saveFileFormat=new SaveFileFormat();
    	$project->save($dataDir . "CreateStandardCalendar.xml", $saveFileFormat->XML);

    	print "Created standard calendar, please check the output file.".PHP_EOL;

    }
}
?>